sqlmesh/core/config/connection.py

```diff
@@ -407,6 +407,7 @@ class SnowflakeConnectionConfig(ConnectionConfig):
     role: t.Optional[str] = None
     authenticator: t.Optional[str] = None
     token: t.Optional[str] = None
+    application: t.ClassVar[str] = "Tobiko_SQLMesh"
```

Comment:
I think doing `t.Literal["Tobiko_SQLMesh"] = "Tobiko_SQLMesh"` is more common in the codebase, but not sure if `t.ClassVar` semantics match the use case better.

Does this guard from the field being mutated, though? Perhaps we should add it to `_FIELD_UPDATE_STRATEGY` with the `IMMUTABLE` strategy? cc @tobymao

Reply:
I'll assume Iaroslav's 👍 means it's worth changing this.